2.3.1 release announcement (#257)
mixxxdj/website: content/news/xxxx-xx-xx-mixxx-2-3-1-released.md

title: Mixxx 2.3.1 released
authors: Daniel Schürmann
tags: 2.3, 2.3.1, release announcement
status: draft
comments: yes

> We are happy to announce first minor release of [Mixxx 2.3.1](https://github.com/mixxxdj/mixxx/releases/tag/2.3.1)!

Reviewer: Isn't 2.3 the minor release? 

```suggestion
We are happy to announce first bugfix release of [Mixxx 2.3.1](https://github.com/mixxxdj/mixxx/releases/tag/2.3.1)!
```

> You will find the download links [here]({filename}/pages/download.md#stable) on Windows, macOS and Linux.

Reviewer: ```suggestion
Head over to the [download page] ({filename}/pages/download.md#stable) and download it on Windows, macOS and Linux.
```

This release comes with new and updated controller mappings, fixes packaging issues and includes some bugfixes.

Mixxx now includes mappings for the [Numark DJ2GO2 Touch](https://manual.mixxx.org/2.3/en/hardware/controllers/numark_dj2go2_touch.html) controller and the [Numark Mixtrack Pro FX](https://manual.mixxx.org/2.3/en/hardware/controllers/numark_mixtrack_pro_fx.html) controller. The [Behringer DDM4000](https://manual.mixxx.org/2.3/en/hardware/controllers/behringer_ddm4000.html), the [Hercules Inpulse 300](https://manual.mixxx.org/2.3/en/hardware/controllers/hercules_djcontrol_inpulse_300.html) mapping, the [Denon MC7000](https://manual.mixxx.org/2.3/en/hardware/controllers/denon_mc7000.html) mapping and the [Traktor S3](https://manual.mixxx.org/2.3/en/hardware/controllers/native_instruments_traktor_kontrol_s3.html) mapping have been improved with Mixxx 2.3.

> Among other minor issue this release fixes the random skipping of tracks and the wrong track entries in the History list when using Auto-DJ, regressions since Mixxx 2.3.0.

Reviewer: ```suggestion
Among other minor issues, this release fixes the random skipping of tracks and the wrong track entries in the history playlist when using AutoDJ.
Both bugs are regressions that were introduced in the Mixxx 2.3.0 release.
```

> For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/changelog.html) and the [2.3.1 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.1).

Reviewer: ```suggestion
For the full list of changes, have a look at the [changelog](https://github.com/mixxxdj/mixxx/blob/2.3.1/CHANGELOG.md) and the [2.3.1 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.1).
```

Author: Mm. The Taged version shows still (unreleased) and the website as well. I have added the date after tagging directly in in the 2.3 branch. 

I think we should do it in future just before the tagging. 
I will update the checklist accordingly. 

What need to be done to update the changelog in the manual? I think I will update it manual.

Author: Ah the release date has arrived automatically. So I think lining the manual is a good idea here. 
Than new users are at the right place for further readings.

Reviewer: > Ah the release date has arrived automatically. So I think lining the manual is a good idea here. 
> Than new users are at the right place for further readings.

But this link does not point to the changelog for 2.3.1. As soon as we release a new mixxx version (e. g. 2.3.2), it points to that changelog. But since this is the 2.3.1 release announcement, it should always point to the 2.3.1 changelog.

Author: The tagged version of the Changelog has the prominent "(Unreleased)" headline. That feels ugly. 
If we link the 2.3 branch it points always to the most recent document, including the latest dot release all with the proper release dates. That's why I like to keep the link to the manual.

Both is working for me, I just want to pint out the implications.

What do you think?

Reviewer: > The tagged version of the Changelog has the prominent "(Unreleased)" headline. That feels ugly.

True. IIRC we already discussed that this "(unreleased)" suffix is unnecessary and confusing when we released 2.3.0. I don't know why it was added this time.

Anyway, I just opened a PR to ensure permanent manual link fragments without "unreleased" or the release date in them. Please have a look: https://github.com/mixxxdj/manual/pull/436

> If we link the 2.3 branch it points always to the most recent document, including the latest dot release all with the proper release dates. That's why I like to keep the link to the manual.

If I read the announcement for the 2.3.1 release and click the changelog link, I expect to be taken to the 2.3.1 changelog, not 2.3.2 or 2.3.3. I suggest to quickly merge mixxxdj/manual#436 and then use a non-changing link suffix. If you don't want further delays, you can also merge this announcement as-is though.

Reviewer: Ah, you already merged it :-)

Now this should work:

```suggestion
For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/changelog.html#v2-3-1) and the [2.3.1 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.1).
```

Author: Cool, thank you.

> In case your are a Linux package maintainer please double check the [Packaging](https://manual.mixxx.org/2.3/en/chapters/appendix/changelog.html#packaging) chapter in the changelog and modify your packaging script if necessary.

Reviewer: ```suggestion
In case you're a Linux package maintainer please double check the [Packaging section](https://github.com/mixxxdj/mixxx/blob/2.3.1/CHANGELOG.md#packaging) in the changelog and modify your packaging script if necessary.
```
